scripts/bin/auto-mount

```diff
@@ -1,17 +1,50 @@
-#! /bin/sh
-
-sleep 5
-
-$SNAP/usr/local/bin/udisksctl monitor | grep --line-buffered " Device:" | while read -r -- device; do
-
-  echo "$SNAP_NAME.auto-mount found ${device}"
-  if echo $device| grep -q "[0-9]$"; then
-    DEV="$(echo $device | sed 's/^Device: //')"
-  fi
-  if [ -n "$DEV" ]; then
-    RES=`$SNAP/usr/local/bin/udisksctl mount --no-user-interaction --block-device "$DEV"`
-    # process returned value into the mount point string
-    RES2=`echo $RES | cut -f4 -d\  | cut -f1 -d.`
-    echo $RES2 > $WATCHED
-  fi
-done
+#!/bin/sh
+
+. "$SNAP/bin/common"
+
+watch_devices() {
+    inotifywait --quiet --monitor --event "moved_to" --event "delete" /dev/disk/by-uuid | \
+        while read -r _parent _action _uuid; do
+            case "$_action" in
+                MOVED_TO) try_mount "$_parent" "$_uuid" ;;
+                DELETE)   rm_mount  "$_uuid" ;;
+                *) : ;;
+            esac
+        done
+}
+
+try_mount() {
+    _parent="${1%/}"
+    _uuid="$2"
+
+    _source="${_parent}/${_uuid}"
+    _target="${_mount_dir}/${_uuid}"
+
+    log "$SNAP_NAME.auto-mount found: ${_source}"
+
+    mkdir -p "$_target"
+    if ! mount "$_source" "$_target" >/dev/null 2>&1; then
+        log "WARN: Could not mount ${_source} to ${_target}"
+        return
+    fi
+
+    echo "$_target" >> "$WATCH_FILE"
+}
+
+rm_mount() {
+    # TODO
```

Comment:
I think `rm_mount()` is just trivially `umount $_target && shutdown {-h|-H $message}`, but of course the Store team recently made the required interface for powering off a system super-privileged. So I think we may just scrap the method of notifying...

Reply:
Sure, if you want to be *safe about it* :P 

There's no down-side to cleaning up after ourselves even if it's just in `/tmp`; it's probably the correct move because (as you said) `MOUNT_DIR` could be set somewhere else.